src/app/file-panel.js

```diff
@@ -108,14 +112,16 @@ function filepanel (appAPI, files) {
 
   var events = new EventManager()
   var element = template()
-  // TODO please do not add custom javascript objects, which have no
-  // relation to the DOM to DOM nodes
-  element.events = events
+
   fileExplorer.events.register('focus', function (path) {
     appAPI.switchToFile(path)
   })
 
-  return element
+  var api = { events: events }
+
+  container.appendChild(element)
+
+  return api
```

Comment:
And because I kind of feel this is related to the issue in the previous versions of this pull request regarding the "API" of modules/component, here again the excerpt from the outdated comment above:

---
Hmm, ...
The object, with the property `api` existed before, but the object was a DOM node.
I wonder how exactly would we want to use it, because that's what we will do all over the place later...

```js
var foobarAPI = { baz: function bar (a,b,c) { console.log(a,b,c) } }
var foobarEvents = files.events // for example
var foobar = new Foobar('.foobarContainer', foobarAPI, foobarEvents)

// the above - as far as i get it - is currently our standard of designing "components", right?
// ... now what is `foobar` ?

// 1. `foobar` was a DOM element with the properties:
var foobar = yo`<div></div>` // but returned from the constructor above of course
foobar.events
foobar.api
foobar.api.callAFoobarMethod()
foobar.events.register('m3h', function () { console.log('meow') }

// 2. then we switched to a model, where a container node is passed into the constructor
var foobar = {
  events: new EventManager(),
  callAFoobarMethod: function () {},
  callAnotherMethod: function () {}
}

// 3. i understand your above proposal as:
var foobar = {
  events: new EventManager(),
  api: {
    callAFoobarMethod: function () {},
    callAnotherMethod: function () {}
  }
}

// 4. Can I add another option that maybe would spontaneously be my favourite option? :P
var foobar = new EventEmitter() // which - by convention - restricts api methods to be called:
// foobar.register
// foobar.unregister
// foobar.trigger
foobar.callAFoobarMethod = function () {}
foobar.callAnotherMethod = function () {}
// personally i would even prefer a system where the `.trigger` method is seperated from the "event emitter", thus one object where callbacks can be registered/unregistered and one on which events can be triggered... the one where events can be triggered should probably stay private to an instance, while the register/unregister method will be public on the instance returned from the constructor call.
```

**What's the reasoning why would any of those options from above be better/worse than others?**

I can see a lot of value in the last option (4.) and i am relatively indifferent to options (1. - 3.)

Whatever the decision would be, it should probably be documented somewhere as a convention for modules/components in general